client/my-sites/migrate/section-migrate.jsx

```diff
@@ -95,16 +95,18 @@ class SectionMigrate extends Component {
 			}
 		} );
 
-		wpcom.undocumented().themes( this.props.sourceSite.ID, { apiVersion: '1' }, ( err, data ) => {
-			if ( data.themes ) {
-				const sourceSiteThemes = [
-					// Put active theme first
-					...data.themes.filter( ( theme ) => theme.active ),
-					...data.themes.filter( ( theme ) => ! theme.active ),
-				];
-				this.setState( { sourceSiteThemes } );
-			}
-		} );
+		wpcom.req
+			.get( `/sites/${ this.props.sourceSite.ID }/themes`, { apiVersion: '1' } )
+			.then( ( data ) => {
+				if ( data.themes ) {
+					const sourceSiteThemes = [
+						// Put active theme first
+						...data.themes.filter( ( theme ) => theme.active ),
+						...data.themes.filter( ( theme ) => ! theme.active ),
```

Comment:
Interesting. In any case, this is something I'd like to address in a follow-up PR, hope that works for you.

Reply:
Noting that this appears to work for me in my testing, and I see the `active` field on themes:

![](https://cldup.com/8-L5LajuXZ.png)

Perhaps you were looking at the response of a different endpoint?

Reply:
It's probably because I was originally looking at a response for a Simple site. That returns 132 WP.com themes, and no `active` field. But for an Atomic site, the response is different: it's just the themes installed on the site, and the `active` field is there. And it's not only the `active` field, the two sets of fields are very different from each other.

Could that be an issue? Is this `section-migrate` UI guaranteed to be shown only for Jetpack/Atomic sites?

Reply:
Well, you can't migrate a WP.com simple site, you'll get this message:

> This site is already hosted on WordPress.com

So I'd say we're safe on this one? Unless I'm misunderstanding.

Reply:
Yes, it's safe then. What confused me is that the responses for Simple and Atomic sites are quite different.